pep8.py

```diff
@@ -1846,7 +1846,15 @@ def check_files(self, paths=None):
         report.start()
         try:
             for path in paths:
-                if os.path.isdir(path):
+                if path == "-":
+                    # STDIN
+                    path = self.options.stdin_name
+                    # if no ersatz filename for STDIN, assume not excluded
+                    excluded = path and self.excluded(path)
+                    if not excluded:
+                        runner(path or "stdin",
+                               lines=stdin_get_value().splitlines(True))
```

Comment:
I'm curious about this too. Why are you keeping the line endings. Plus, why is this being special cased here?